src/rgw/rgw_sal_motr.cc

```diff
@@ -2730,6 +2730,13 @@ int MotrObject::update_null_reference(const DoutPrefixProvider *dpp, rgw_bucket_
   rgw_bucket_dir_entry current_null_key_ref;
   current_null_key_ref.key.name = this->get_name();
   current_null_key_ref.key.instance = this->get_instance();
+
+  // Set the key and instance for multipart object from ent structure
+  if (ent.meta.category == RGWObjCategory::MultiMeta)
```

Comment:
well why not putting lines 2371-2732 & 2737-2738 in if-else condition

Reply:
sure, done.